File: app/src/main/java/com/example/s_tools/Splash_login_reg/SplashScreen.java
package com.example.s_tools.Splash_login_reg;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.view.animation.DecelerateInterpolator;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.ScrollView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.core.view.ViewCompat;
import androidx.core.view.ViewPropertyAnimatorCompat;

import com.example.s_tools.R;
import com.example.s_tools.Splash_login_reg.RegisterApiCall.RegisterNewAccount;
import com.example.s_tools.Splash_login_reg.loginapi_call.LoginApiCall;
import com.example.s_tools.entertainment.VideoActivity;
import com.example.s_tools.tools.ToastMy;
import com.example.s_tools.tools.Validation;
import com.example.s_tools.tools.retrofitcalls.VC;
import com.kaopiz.kprogresshud.KProgressHUD;

public class SplashScreen extends AppCompatActivity {
    public static final int STARTUP_DELAY = 300;
    public static final int ITEM_DELAY = 300;
    public static final String TAG="mtag";
    public static final String INVALID_NAME="Invalid Name";
    public static final String INVALID_USERNAME="Invalid username";
    public static final String WAIT="Please wait..";
    public static final String REGISTER="REGISTER";
    public static final String LOGIN="LOGIN";
    EditText name,username,email,pass;
    ScrollView scrollView;
    ImageView logo;
    Button register,back_btn;
    CardView login,registerok;
    TextView registerHint,nootwelcomemsg,statusemail,statuspass,logintxt,registertxt;
    LinearLayout hintLayout;
    ProgressBar loginprogress,registerprogress;
    TextView taphere;
    View line;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        overridePendingTransition(R.anim.fadein,R.anim.fadeout);
        setContentView(R.layout.splashscreen_bg);
        init();
        this.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
        animate();

        new Handler().postDelayed(this::setupListner,1000);
    }

    private void setupListner() {
        login.setOnClickListener(v -> {
            statuspass.setText("");
            if (pass.getText().toString().isEmpty() || pass.getText().toString().contains(" ")){
                statuspass.setText("Not Valid Password");
                return;
            }else if (email.getText().toString().isEmpty() ||!Validation.isValidEmailId(email.getText().toString(), SplashScreen.this)){
                return;
            }
            logintxt.setText(WAIT);
            loginprogress.setVisibility(View.VISIBLE);
            hideKeyboard();
            LoginApiCall.login(SplashScreen.this, email.getText().toString(), pass.getText().toString(), success -> {
                loginprogress.setVisibility(View.GONE);
                if (!success){
                    logintxt.setText(LOGIN);
                }else {
                    putAppData();
                }
            });
        });


        //successfull register
        registerok.setOnClickListener(v -> {

            boolean validPassword=Validation.isValidPassword(pass.getText().toString(), SplashScreen.this);
            boolean validEmailId=Validation.isValidEmailId(email.getText().toString().trim(), SplashScreen.this);


            if (validEmailId && validPassword){
                if(name.getText().toString().isEmpty() || name.getText().toString().length() < 3 || name.getText().toString().contains(" ")){
                    Toast.makeText(SplashScreen.this, INVALID_NAME, Toast.LENGTH_SHORT).show();
                    registertxt.setText(REGISTER);
                    registerprogress.setVisibility(View.GONE);
                    return;
                }
                if (username.getText().toString().isEmpty() || username.getText().toString().length() < 4 || username.getText().toString().contains(" ")){
                    Toast.makeText(SplashScreen.this, INVALID_USERNAME, Toast.LENGTH_SHORT).show();
                    registertxt.setText(REGISTER);
                    registerprogress.setVisibility(View.GONE);
                    return;
                }
                hideKeyboard();
                registertxt.setText(WAIT);
                registerprogress.setVisibility(View.VISIBLE);

                //register api call
                RegisterNewAccount.registerNew(SplashScreen.this, name.getText().toString(), username.getText().toString(), email.getText().toString(), pass.getText().toString(), null, new RegisterNewAccount.ApiCallback() {
                    @Override
                    public void onResponse(boolean success) {
                        if (!success){
                            registertxt.setText(REGISTER);
                            registerprogress.setVisibility(View.GONE);
                        }else {
                            LoginApiCall.login(SplashScreen.this, email.getText().toString(), pass.getText().toString(), success1 -> {
                                if (success){
                                    ToastMy.successToast(SplashScreen.this,"Registered Successful", ToastMy.LENGTH_SHORT);
                                    startActivity(new Intent(SplashScreen.this, VideoActivity.class));
                                    finish();
                                }
                            });
                        }
                    }
                });
            }

        });

        //register
        register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                animate();
                line.setVisibility(View.VISIBLE);
                registerHint.setVisibility(View.VISIBLE);
                logo.setVisibility(View.GONE);
                nootwelcomemsg.setVisibility(View.GONE);
                name.setVisibility(View.VISIBLE);
                hintLayout.setVisibility(View.GONE);
                registerok.setVisibility(View.VISIBLE);
                back_btn.setVisibility(View.VISIBLE);
                register.setVisibility(View.GONE);
                login.setVisibility(View.GONE);
                username.setVisibility(View.VISIBLE);
                logo.getLayoutParams().height = 230;
                logo.getLayoutParams().width = 230;
            }
        });
        //back btn
        back_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                line.setVisibility(View.GONE);
                registerHint.setVisibility(View.GONE);
                logo.setVisibility(View.VISIBLE);
                nootwelcomemsg.setVisibility(View.VISIBLE);
                name.setVisibility(View.GONE);
                hintLayout.setVisibility(View.VISIBLE);
                registerok.setVisibility(View.GONE);
                back_btn.setVisibility(View.GONE);
                register.setVisibility(View.VISIBLE);
                login.setVisibility(View.VISIBLE);
                username.setVisibility(View.GONE);
                logo.getLayoutParams().height = 260;
                logo.getLayoutParams().width = 260;
            }
        });
        taphere.setOnClickListener(view -> email.setText(email.getText().append("@gmail.com")));
    }

    private void putAppData() {
        KProgressHUD kProgressHUD=KProgressHUD.create(SplashScreen.this).setStyle(KProgressHUD.Style.SPIN_INDETERMINATE).setLabel("Please Wait...").setDimAmount(0.5f)
                .setCancellable(false).show();
        VC.check(SplashScreen.this, (success, updateinfo) -> {
            if (success){
                kProgressHUD.dismiss();
                startActivity(new Intent(SplashScreen.this,VideoActivity.class));
                finish();
            }
        });
    }

    private void hideKeyboard() {
        InputMethodManager imm = (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
        imm.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0);
    }

    private void init() {
        line = findViewById(R.id.lineview);
        taphere = findViewById(R.id.taphere);
        registerHint = findViewById(R.id.registerhint);
        name = (EditText)findViewById(R.id.name_edittext);
        username = (EditText)findViewById(R.id.username_editext);
        email = (EditText)findViewById(R.id.email);
        pass = (EditText)findViewById(R.id.password);

        statuspass = (TextView) findViewById(R.id.status_password);
        nootwelcomemsg = (TextView) findViewById(R.id.noot_welcome_you);
        hintLayout = (LinearLayout)findViewById(R.id.hint_linearLayout);

        scrollView = (ScrollView)findViewById(R.id.mscrollView);
        logo = (ImageView)findViewById(R.id.img_logo);
        register = (Button)findViewById(R.id.register_btn);
        registerok = (CardView) findViewById(R.id.register_ok);
        back_btn = (Button)findViewById(R.id.back_btn_registrationform);

        login = (CardView)findViewById(R.id.login_btn);
        loginprogress = findViewById(R.id.login_ok_progressbar);
        registerprogress=findViewById(R.id.register_ok_progressbar);
        logintxt = findViewById(R.id.login_txt);
        registertxt= findViewById(R.id.register_ok_txt);
    }
    private void animate() {
        ImageView logoImageView = (ImageView) findViewById(R.id.img_logo);
        ViewGroup container = (ViewGroup) findViewById(R.id.container);

        ViewCompat.animate(logoImageView)
                .translationY(-150)
                .setStartDelay(STARTUP_DELAY)
                .setDuration(1000).setInterpolator(
                new DecelerateInterpolator(2.0f)).start();

        for (int i = 0; i < container.getChildCount(); i++) {
            View v = container.getChildAt(i);
            ViewPropertyAnimatorCompat viewAnimator;

            if (!(v instanceof Button)) {
                viewAnimator = ViewCompat.animate(v)
                        .translationY(0).alpha(1)

                        .setStartDelay((ITEM_DELAY)+200)
                        .setDuration(500);
            } else {
                viewAnimator = ViewCompat.animate(v)
                        .scaleY(1).scaleX(1)
                        .setStartDelay((ITEM_DELAY)+400)
                        .setDuration(500);
            }

            viewAnimator.setInterpolator(new DecelerateInterpolator()).start();
        }
    }
}

File: app/src/main/java/com/example/s_tools/Splash_login_reg/SplashScreen.kt
package com.example.s_tools.Splash_login_reg

import com.example.s_tools.Splash_login_reg.loginapi_call.LoginApiCall.login
import com.example.s_tools.Splash_login_reg.RegisterApiCall.RegisterNewAccount.registerNew
import com.example.s_tools.tools.retrofitcalls.VC.check
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import android.os.Bundle
import com.example.s_tools.R
import android.view.WindowManager
import com.example.s_tools.tools.Validation
import com.example.s_tools.Splash_login_reg.loginapi_call.LoginApiCall
import com.example.s_tools.Splash_login_reg.RegisterApiCall.RegisterNewAccount
import com.example.s_tools.tools.ToastMy
import android.content.Intent
import com.example.s_tools.entertainment.VideoActivity
import com.kaopiz.kprogresshud.KProgressHUD
import com.example.s_tools.tools.retrofitcalls.VC
import android.app.Activity
import android.os.Handler
import android.view.View
import android.view.ViewGroup
import android.view.animation.DecelerateInterpolator
import android.view.inputmethod.InputMethodManager
import android.widget.*
import androidx.core.view.ViewCompat
import androidx.core.view.ViewPropertyAnimatorCompat

class SplashScreen() : AppCompatActivity() {
    var name: EditText? = null
    var username: EditText? = null
    var email: EditText? = null
    var pass: EditText? = null
    var scrollView: ScrollView? = null
    var logo: ImageView? = null
    var register: Button? = null
    var back_btn: Button? = null
    var login: CardView? = null
    var registerok: CardView? = null
    var registerHint: TextView? = null
    var nootwelcomemsg: TextView? = null
    var statusemail: TextView? = null
    var statuspass: TextView? = null
    var logintxt: TextView? = null
    var registertxt: TextView? = null
    var hintLayout: LinearLayout? = null
    var loginprogress: ProgressBar? = null
    var registerprogress: ProgressBar? = null
    var taphere: TextView? = null
    var line: View? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        overridePendingTransition(R.anim.fadein, R.anim.fadeout)
        setContentView(R.layout.splashscreen_bg)
        init()
        this.window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)
        animate()
        Handler().postDelayed({ setupListner() }, 1000)
    }

    private fun setupListner() {
        login!!.setOnClickListener { v: View? ->
            statuspass!!.setText("")
            if (pass!!.getText().toString().isEmpty() || pass!!.getText().toString()
                    .contains(" ")
            ) {
                statuspass!!.setText("Not Valid Password")
                return@setOnClickListener
            } else if (email!!.getText().toString().isEmpty() || !Validation.isValidEmailId(
                    email!!.getText().toString(),
                    this@SplashScreen
                )
            ) {
                return@setOnClickListener
            }
            logintxt!!.setText(WAIT)
            loginprogress!!.setVisibility(View.VISIBLE)
            hideKeyboard()
            login(
                this@SplashScreen,
                email!!.getText().toString(),
                pass!!.getText().toString()
            ) { success: Boolean ->
                loginprogress!!.setVisibility(View.GONE)
                if (!success) {
                    logintxt!!.setText(LOGIN)
                } else {
                    putAppData()
                }
            }
        }


        //successfull register
        registerok!!.setOnClickListener { v: View? ->
            val validPassword: Boolean = Validation.isValidPassword(
                pass!!.getText().toString(), this@SplashScreen
            )
            val validEmailId: Boolean = Validation.isValidEmailId(
                email!!.getText().toString().trim { it <= ' ' },
                this@SplashScreen
            )
            if (validEmailId && validPassword) {
                if (name!!.getText().toString().isEmpty() || (name!!.getText()
                        .toString().length < 3) || name!!.getText().toString().contains(" ")
                ) {
                    Toast.makeText(this@SplashScreen, INVALID_NAME, Toast.LENGTH_SHORT).show()
                    registertxt!!.setText(REGISTER)
                    registerprogress!!.setVisibility(View.GONE)
                    return@setOnClickListener
                }
                if (username!!.getText().toString().isEmpty() || (username!!.getText()
                        .toString().length < 4) || username!!.getText().toString().contains(" ")
                ) {
                    Toast.makeText(this@SplashScreen, INVALID_USERNAME, Toast.LENGTH_SHORT).show()
                    registertxt!!.setText(REGISTER)
                    registerprogress!!.setVisibility(View.GONE)
                    return@setOnClickListener
                }
                hideKeyboard()
                registertxt!!.setText(WAIT)
                registerprogress!!.setVisibility(View.VISIBLE)

                //register api call
                registerNew(
                    this@SplashScreen,
                    name!!.getText().toString(),
                    username!!.getText().toString(),
                    email!!.getText().toString(),
                    pass!!.getText().toString(),
                    null,
                    object : RegisterNewAccount.ApiCallback {
                        override fun onResponse(success: Boolean) {
                            if (!success) {
                                registertxt!!.setText(REGISTER)
                                registerprogress!!.setVisibility(View.GONE)
                            } else {
                                login(
                                    this@SplashScreen,
                                    email!!.getText().toString(),
                                    pass!!.getText().toString()
                                ) { success1: Boolean ->
                                    if (success) {
                                        ToastMy.successToast(
                                            this@SplashScreen,
                                            "Registered Successful",
                                            ToastMy.LENGTH_SHORT
                                        )
                                        startActivity(
                                            Intent(
                                                this@SplashScreen,
                                                VideoActivity::class.java
                                            )
                                        )
                                        finish()
                                    }
                                }
                            }
                        }
                    })
            }
        }

        //register
        register!!.setOnClickListener(View.OnClickListener { //                animate();
            line!!.visibility = View.VISIBLE
            registerHint!!.visibility = View.VISIBLE
            logo!!.visibility = View.GONE
            nootwelcomemsg!!.visibility = View.GONE
            name!!.visibility = View.VISIBLE
            hintLayout!!.visibility = View.GONE
            registerok!!.visibility = View.VISIBLE
            back_btn!!.visibility = View.VISIBLE
            register!!.visibility = View.GONE
            login!!.visibility = View.GONE
            username!!.visibility = View.VISIBLE
            logo!!.layoutParams.height = 230
            logo!!.layoutParams.width = 230
        })
        //back btn
        back_btn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                line!!.visibility = View.GONE
                registerHint!!.visibility = View.GONE
                logo!!.visibility = View.VISIBLE
                nootwelcomemsg!!.visibility = View.VISIBLE
                name!!.visibility = View.GONE
                hintLayout!!.visibility = View.VISIBLE
                registerok!!.visibility = View.GONE
                back_btn!!.visibility = View.GONE
                register!!.visibility = View.VISIBLE
                login!!.visibility = View.VISIBLE
                username!!.visibility = View.GONE
                logo!!.layoutParams.height = 260
                logo!!.layoutParams.width = 260
            }
        })
        taphere!!.setOnClickListener({ view: View? ->
            email!!.setText(
                email!!.getText().append("@gmail.com")
            )
        })
    }

    private fun putAppData() {
        val kProgressHUD =
            KProgressHUD.create(this@SplashScreen).setStyle(KProgressHUD.Style.SPIN_INDETERMINATE)
                .setLabel("Please Wait...").setDimAmount(0.5f)
                .setCancellable(false).show()
        check(this@SplashScreen) { success: Boolean, updateinfo: String? ->
            if (success) {
                kProgressHUD.dismiss()
                startActivity(Intent(this@SplashScreen, VideoActivity::class.java))
                finish()
            }
        }
    }

    private fun hideKeyboard() {
        val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
        imm.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0)
    }

    private fun init() {
        line = findViewById(R.id.lineview)
        taphere = findViewById(R.id.taphere)
        registerHint = findViewById(R.id.registerhint)
        name = findViewById<View>(R.id.name_edittext) as EditText
        username = findViewById<View>(R.id.username_editext) as EditText
        email = findViewById<View>(R.id.email) as EditText
        pass = findViewById<View>(R.id.password) as EditText
        statuspass = findViewById<View>(R.id.status_password) as TextView
        nootwelcomemsg = findViewById<View>(R.id.noot_welcome_you) as TextView
        hintLayout = findViewById<View>(R.id.hint_linearLayout) as LinearLayout
        scrollView = findViewById<View>(R.id.mscrollView) as ScrollView
        logo = findViewById<View>(R.id.img_logo) as ImageView
        register = findViewById<View>(R.id.register_btn) as Button
        registerok = findViewById<View>(R.id.register_ok) as CardView
        back_btn = findViewById<View>(R.id.back_btn_registrationform) as Button
        login = findViewById<View>(R.id.login_btn) as CardView
        loginprogress = findViewById(R.id.login_ok_progressbar)
        registerprogress = findViewById(R.id.register_ok_progressbar)
        logintxt = findViewById(R.id.login_txt)
        registertxt = findViewById(R.id.register_ok_txt)
    }

    private fun animate() {
        val logoImageView = findViewById<View>(R.id.img_logo) as ImageView
        val container = findViewById<View>(R.id.container) as ViewGroup
        ViewCompat.animate(logoImageView)
            .translationY(-150f)
            .setStartDelay(STARTUP_DELAY.toLong())
            .setDuration(1000).setInterpolator(
                DecelerateInterpolator(2.0f)
            ).start()
        for (i in 0 until container.childCount) {
            val v = container.getChildAt(i)
            var viewAnimator: ViewPropertyAnimatorCompat
            if (v !is Button) {
                viewAnimator = ViewCompat.animate(v)
                    .translationY(0f).alpha(1f)
                    .setStartDelay(((ITEM_DELAY) + 200).toLong())
                    .setDuration(500)
            } else {
                viewAnimator = ViewCompat.animate(v)
                    .scaleY(1f).scaleX(1f)
                    .setStartDelay(((ITEM_DELAY) + 400).toLong())
                    .setDuration(500)
            }
            viewAnimator.setInterpolator(DecelerateInterpolator()).start()
        }
    }

    companion object {
        val STARTUP_DELAY = 300
        val ITEM_DELAY = 300
        val TAG = "mtag"
        val INVALID_NAME = "Invalid Name"
        val INVALID_USERNAME = "Invalid username"
        val WAIT = "Please wait.."
        val REGISTER = "REGISTER"
        val LOGIN = "LOGIN"
    }
}